### PR TITLE
[5.x] Fix `JobWatcher` when processing deleted instance of `SerializesModels`

### PR DESCRIPTION
**Pull request by hapidjus**
If there is a failure when unserializing the command, (for instance if the job deleted a model it received,  of the database connection is set dynamically) the jobs will never proceed to processed. See #1479

In this PR unserialize is removed in favor of preg_replace to extract the batch_id

### CODE REVIEW

**Comment by crynobone** [comment]
@hapidjus shouldn't we use the new approach only if `unserialize()` return an exception?

**Comment by hapidjus** [comment]
> @hapidjus shouldn't we use the new approach only if `unserialize()` return an exception?

Makes sense. 
This PR was mainly to confirm that using preg_match is a reasonable approach here.
The PR currently doesn’t handle encrypted data or include any tests.

However I'm not sure we can catch exceptions that happens "inside" the unserialize function. It's not necessarily unserialize() that throws.

**Comment by hapidjus** [comment]
I've updated the PR now to try to unserialize first and only use preg_match as fallback.

**Comment by crynobone** [comment]
@hapidjus Thanks, I'll review it and possibility add some tests to cover the new use case. Will mark it as ready when I'm done with it

File: src/Watchers/JobWatcher.php [hunk 1]
```diff
@@ -266,7 +267,7 @@ protected function getBatchId(array $data)
             $properties = ExtractProperties::from($command);
 
             return $properties['batchId'] ?? null;
-        } catch (\Exception|\Error) {
```
Comment (hapidjus) [inline src/Watchers/JobWatcher.php]: I understrand that `catch (\Exception|\Error)` is a bit greedy. But there are other ways that the unserialize can go wrong. For instance when in a multi tenancy context and the database connection has not been set yet. See this issue for instance https://github.com/archtechx/tenancy/issues/1263

Comment (crynobone) [reply src/Watchers/JobWatcher.php]: If there a use case to include other exceptions we need to see the actual use case and replication steps so we can add tests to cover that usage. Having the scope too wide add a huge risk to this PR getting rejected.

Comment (hapidjus) [reply src/Watchers/JobWatcher.php]: Alright, I'll see if I can make a minimal reproduction of it. Bur for now I'm happy to have this merged and make another PR later.

**Pull request merged**